sdk/python/src/dagger/client/_session.py

```diff
@@ -172,6 +172,7 @@ def __await__(self):
         return self.connect().__await__()
 
     async def __aenter__(self) -> Self:
+        telemetry.initialize()
```

Comment:
The initialization was setup initially while the SDK establishes a connection to the API, but it was added only to the connection that has automatic provisioning support. So added to this one too, which is the one used in modules.